src/kernel.cr

```diff
@@ -217,23 +217,31 @@ end
 # Flag     | Applies to    | Meaning
 # ---------+---------------+--------------------------------------------
 # space    | bdiouxX       | Add a leading space character to
-#          | aAeEfgG       | non-negative numbers.
-#          | (numeric fmt) | For o, x, X, b, use
-#          |               | a minus sign with absolute value for
-#          |               | negative values.
+#          | aAeEfgG       | non-negative numbers. Has no effect if the plus
+#          | (numeric fmt) | flag is also given.
+# ---------+---------------+-----------------------------------------
+# #        | boxX          | Use an alternative format.
+#          | aAeEfgG       | For o, x, X, b, prefix any non-zero result with
+#          |               | 0o, 0x, 0X, or 0b respectively.
+#          |               | For a, A, e, E, f, g, G,
+#          |               | force a decimal point to be added,
+#          |               | even if no digits follow.
+#          |               | For g and G, do not remove trailing zeros.
```

Comment:
I think we're missing specs for this.

Reply:
This part of the documentation is simply copied from C because our implementation indeed delegates to `LibC.snprintf`. This PR only deals with integers so the float specs are moved verbatim from `string_spec.cr` and no new specs are added, but it probably doesn't hurt to use `#` at least once.

Reply:
Oh the method actually raises `Non-resizeable stream` now...

Reply:
(or rather, `"%#.1g" % 123.0` currently raises already, but now is a good opportunity to fix it)